src/app/checkout/step-3/step-3.component.spec.js

```diff
@@ -474,7 +349,34 @@ describe('checkout', () => {
       it('should call handleRecaptchaFailure if the recaptchaFailedEvent is received', () => {
         jest.spyOn(self.controller, 'handleRecaptchaFailure').mockImplementation(() => {})
         self.controller.$rootScope.$emit(recaptchaFailedEvent)
-        expect(self.controller.handleRecaptchaFailure).toHaveBeenCalledWith(self.controller)
+        expect(self.controller.handleRecaptchaFailure).toHaveBeenCalled();
+      })
+    })
+
+    describe('submitOrder', () => {
+      beforeEach(() => {
+        jest.spyOn(self.controller.analyticsFactory, 'purchase')
+        self.controller.retrieveCoverFeeDecision = () => true;
+      })
+
+      it('should call analyticsFactory when it is not branded checkout', () => {
+        jest.spyOn(self.controller.orderService, 'retrieveCoverFeeDecision')
```

Comment:
You can remove this line and the same line in the next test, and add `jest.spyOn(self.controller.orderService, 'retrieveCoverFeeDecision').mockReturnValue(true)` to the `beforeEach` block.